docs/proposals/05-etcd-operator-tasks.md

```diff
@@ -0,0 +1,396 @@
+
+## Summary
+
+The proposal suggests the creation of a unified framework for managing "out-of-band" operator tasks in an etcd cluster. It recommends using a single dedicated custom resource for all "out-of-band" operator tasks, simplifying the process of adding new tasks and managing their lifecycle. The proposal also suggests using a single controller to orchestrate these tasks, initiating actions based on the task type specified in the custom resource.
+The custom resource(say EtcdOperatorTasks), will have specific fields in its spec (desired state) and status (current state) to capture the type and configuration of each task, as well as their status and any errors.
+The proposal also addresses the challenge of dealing with multiple or duplicate tasks triggered simultaneously. It suggests defining pre-conditions for each task, which must be met for the task to be executed. If multiple tasks are pending, they will be executed in a First-In-First-Out (FIFO) manner.
+
+## Terminology
+
+* **druid:** [etcd-druid](https://github.com/gardener/etcd-druid) is an operator to manage the etcd clusters.
+
+* **backup-sidecar:** It is the etcd-backup-restore sidecar container running in each etcd-member pod of etcd cluster.
+
+* **leading-backup-sidecar:** A backup-sidecar that is associated to an etcd leader of an etcd cluster.
+
+* **on-demand**: Refers to any task or operation that is performed outside of the regular schedule.
+
+## Motivation
+
+[Etcd-druid](https://github.com/gardener/etcd-druid) mainly acts as an etcd cluster provisioner(creation, maintenance and deletion), but sometimes operator wants to perform some on-demand operations on etcd cluster. These operations, which can be executed on-demand, are referred to as "out-of-band" operations/tasks. Operations that have nothing to do with the desired state of the etcd cluster.
+Today, these "out-of-band" task, operator have to perform via different methods like manual, or via some other ways, but there is no single standard interface available for operator to use that for these "out-of-band" operator task
+This proposal proposes a unified framework to perform operator's `out-of-band` operations/tasks on etcd cluster.
+
+Examples of various "out-of-band" operator task are following:
+
+* Recovery from permanent quorum loss.
+* Trigger on-demand full/delta snapshots.
+* Trigger on-demand snapshot compaction.
+* Trigger on-demand maintenance of etcd cluster.
+* Copy the backups from one object store to another object store.
+
+## Definition of "out-of-band" task
+
+"out-of-band" task refer to any on-demand tasks/operations that can be executed on an etcd cluster without modifying the Etcd custom resource spec (desired state).
+
+## Goals
+
+* Establish a unified interface for operators to initiate any "out-of-band" operator task at their convenience and monitor their status.
+* Facilitate the easy addition of new "out-of-band" task through this framework.
+
+### Concrete Use Cases
```

Comment:
As discussed. Start make some structural changes to the overall document. 
Proposal:
- API
  - Custom Resource Object
  - Golang API
- Lifecycle
  - Creation
  - Execution
    - Pre-Condition or acceptance criteria for a task
    - Sequencing of task execution
  - Deletion
- Use Cases
  - <Use Case 1>
    - Describe the use case
    - List scenario where this task is going to be used.
    - Define the API
    - Define the pre-conditions

Remove the following sections:
- 1 custom resource vs n custom resources    
- 1 contoller vs n-controller

Replace `Graduation Criteria` with feature gates.